bin/autodetect/Main.cc

```diff
@@ -194,16 +194,16 @@ int main(int argc, char** argv) {
     limits.init(analysisLimits.categorizationExamplesLimit(),
                 analysisLimits.modelMemoryLimitMb());
 
-    bool doingCategorization{fieldConfig.fieldNameSuperset().count(
+    const ml::api::CAnomalyJobConfig::CAnalysisConfig& analysisConfig =
+        jobConfig.analysisConfig();
+
+    bool doingCategorization{analysisConfig.fieldNameSuperset().count(
```

Comment:
It looks like the only use of `fieldNameSuperset` is to check if it contains `mlcategory`.  This way of testing dates back to the days when it was not compulsory to configure `categorization_field_name`, and if you used `mlcategory` in a detector without explicitly specifying `categorization_field_name` then `categorization_field_name` defaulted to `_raw`.  But now it's compulsory to configure `categorization_field_name` if you want categorization you could replace this check with:

```
    bool doingCategorization{analysisConfig.categorizationFieldName().empty() == false};
```

If you do that everywhere where `fieldNameSuperset` is being called then you'll avoid having to populate the set in the first place.

(Ignore this comment if there's some other usage of `fieldNameSuperset` that I missed.)

Reply:
Yes that's correct. This construct is only used to check to see if categorization is wanted - your suggestion is far simpler and makes the desired intent much clearer.